src/stash/build.rs

```diff
@@ -101,49 +101,72 @@ fn main() -> anyhow::Result<()> {
     let hashes = fs::File::open(PROTO_HASHES).context("opening proto hashes")?;
```

Comment:
I think we can fix the rebuild issue if we stick
```
println!("cargo:rerun-if-changed=protos");
```
at the beginning of `main`.

According to the [docs](https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath) the default behavior is to re-run the build script if anything in the package changes, and in this case I'm wondering if it's actually anything in our workspace. By adding the `rerun-if-changed` directive it should only re-run if something in the protos directory changes.